packages/components/addon/components/hds/tabs/index.js

```diff
@@ -1,11 +1,74 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 export default class HdsTabsIndexComponent extends Component {
-  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
-  // constructor() {
-  //   super(...arguments);
-  //   // ADD YOUR ASSERTIONS HERE
-  // }
+  @tracked tabNodes = [];
+  @tracked tabIds = [];
+  @tracked panelNodes = [];
+  @tracked panelIds = [];
+  @tracked selectedTabIndex;
+
+  @action
+  didInsert() {
+    // default starting tab index
+    let initialTabIndex = 0;
+
+    this.tabNodes.forEach((tabElement, index) => {
+      if (tabElement.hasAttribute('data-is-selected')) {
+        initialTabIndex = index;
+      }
+    });
+    this.selectedTabIndex = initialTabIndex;
+  }
+
+  @action
+  didInsertTab(element) {
+    this.tabNodes = [...this.tabNodes, element];
+    this.tabIds = [...this.tabIds, element.id];
```

Comment:
That's what I tried initially but there were some weird behaviors I was getting. The initially selected tab wasn't getting set for some reason. I can try it again in case it was happening from some other part of the earlier version of the code I was working on. (Will look at it again after merging this branch.)